### PR TITLE
feat: builds helm chart

### PR DESCRIPTION
**Pull request by AlexsJones**
#### The second part will be to serve the chart from the repo 

After this is accepted I will add the code to convert this repo into a helm repo which will enable

```
helm repo add ofo https://github.com/open-feature/open-feature-operator
helm upgrade --install release ofo/open-feature-operator
```

for now you can use

```
#clone the git repo 
cd chart
helm upgrade ofo . --install
```

### CODE REVIEW

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -89,6 +89,7 @@ jobs:
             make update-flagd
             go mod tidy
             make controller-gen
+            CHART_VERSION=${{ needs.release-please.outputs.release_tag_name }} make helm-package
```
Comment (AlexsJones) [inline .github/workflows/release-please.yml:92]: I dont know if this will work tbh because of the `v` helm is semvar v2

File: Makefile [hunk 2]
```diff
@@ -153,3 +156,23 @@ $(CONTROLLER_GEN): $(LOCALBIN)
...
 	GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+
+OSARCH=$(shell ./hack/get-os.sh)
+HELM = $(shell pwd)/bin/$(OSARCH)/helm
+HELM_INSTALLER ?= "https://get.helm.sh/helm-v3.10.1-$(OSARCH).tar.gz"
+.PHONY: helm
+helm: $(HELM) ## Download helm locally if necessary.
+$(HELM): $(LOCALBIN)
+	[ -e "$(HELM)" ] && rm -rf "$(HELM)" || true
+	cd $(LOCALBIN) && curl -s $(HELM_INSTALLER) | tar xvf - -C $(LOCALBIN) || tar xvf - -C $(LOCALBIN)
```
Comment (toddbaert) [inline Makefile:167]: This fails for me. Do you mean `tar xzf`? It seems like the file is gzipped, so I'd expect we run `gunzip`.

Comment (AlexsJones) [reply Makefile:167]: This is actually specific to OS.. urgh! 
what are you running on?

Comment (AlexsJones) [reply Makefile:167]: yes, we could use gunzip ill update

Comment (toddbaert) [reply Makefile:167]: > This is actually specific to OS.. urgh!
> what are you running on?

```
                   -`
                  .o+`
                 `ooo/                   todd@vixen 
                `+oooo:                  ---------- 
               `+oooooo:                 OS: Arch Linux x86_64 
               -+oooooo+:                Host: GS66 Stealth 10SE REV:1.0 
             `/:-:++oooo+:               Kernel: 5.19.12-arch1-1 
            `/++++/+++++++:              Uptime: 2 hours, 45 mins 
           `/++++++++++++++:             Packages: 1370 (pacman), 1 (flatpak) 
          `/+++ooooooooooooo/`           Shell: bash 5.1.16 
         ./ooosssso++osssssso+`          Resolution: 1080x1920, 1920x1080, 1920x1080 
        .oossssso-````/ossssss+`         DE: GNOME 42.5 
       -osssssso.      :ssssssso.        WM: Mutter 
      :osssssss/        osssso+++.       WM Theme: Adwaita 
     /ossssssss/        +ssssooo/-       Theme: Adwaita-dark [GTK2/3] 
   `/ossssso+/:-        -:/+osssso+-     Icons: Adwaita [GTK2/3] 
  `+sso+:-`                 `.-/+oso:    Terminal: vscode 
 `++:.                           `-/+/   CPU: Intel i7-10750H (12) @ 5.000GHz 
 .`                                 `/   GPU: NVIDIA GeForce RTX 2060 Mobile 
                                         GPU: Intel CometLake-H GT2 [UHD Graphics] 
                                         Memory: 11234MiB / 31919MiB 
```

BTW

File: .github/workflows/release-please.yml (hunk 1 above)
Comment (toddbaert) [reply .github/workflows/release-please.yml:92]: See https://github.com/open-feature/open-feature-operator/pull/137#discussion_r1000960319

**Comment by toddbaert** [comment]
> for now you can use
> 
> ```
> #clone the git repo 
> cd chart
> helm upgrade --install . ofo 
> ```

I _think_ you mean `helm upgrade ofo --install .` - that worked for me

File: chart/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v2
+name: chart
+description: A Helm chart for Kubernetes
```
Comment (toddbaert) [inline chart/Chart.yaml:3]: Should these be updated?

File: Makefile (hunk 2 above)
Comment (AlexsJones) [reply Makefile]: can you try this now? I think `tar was just missing the `-` flag prefix

Comment (toddbaert) [reply Makefile]: I get the same error:

```$ make helm
[ -e "/home/todd/git/open-feature-operator/bin/linux-amd64/helm" ] && rm -rf "/home/todd/git/open-feature-operator/bin/linux-amd64/helm" || true
cd /home/todd/git/open-feature-operator/bin && curl -s "https://get.helm.sh/helm-v3.10.1-linux-amd64.tar.gz" | tar -xvf - -C /home/todd/git/open-feature-operator/bin
tar: Archive is compressed. Use -z option
tar: Error is not recoverable: exiting now
make: *** [Makefile:167: /home/todd/git/open-feature-operator/bin/linux-amd64/helm] Error 2
```

`tar -xzf` does work for me (gnutar), but I'm not sure if that will break on bsdtar (which is I guess what you're using)

File: chart/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# to be deployed.
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
```
[the remark this answers is not in the document]
Comment (AlexsJones) [reply chart/Chart.yaml:16]: If i do that, people won't be able to deploy the chart from main

File: chart/Chart.yaml (hunk 3 above)
Comment (AlexsJones) [reply chart/Chart.yaml:3]: done

**Comment by AlexsJones** [comment]
> 

Fixed, this was due to missing CRD's and wrong image

File: Makefile (hunk 2 above)
Comment (AlexsJones) [reply Makefile]: that should work fine actually, I think the `c` is bsd specific

**Comment by AlexsJones** [comment]
image

File: .release-please-manifest.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
 {
-  ".": "0.2.6"
+  ".": "0.2.5"
```
Comment (skyerus) [inline .release-please-manifest.json:2]: Is this right?

File: Makefile [hunk 6]
```diff
@@ -3,6 +3,7 @@
 IMG ?= controller:latest
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 FLAGD_VERSION=v0.2.5
+CHART_VERSION=# x-release-please-version
```
Comment (skyerus) [inline Makefile:6]:
```suggestion
CHART_VERSION=v0.2.6 # x-release-please-version
```
I think release-please needs something to replace

File: chart/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "0.1.0"
```
Comment (skyerus) [inline chart/Chart.yaml:21]:
```suggestion
appVersion: "v0.2.6" # x-release-please-version
```

File: .release-please-manifest.json (hunk 5 above)
Comment (AlexsJones) [reply .release-please-manifest.json:2]: fixed

File: Makefile (hunk 6 above)
Comment (AlexsJones) [reply Makefile:6]: updated

File: Makefile [hunk 8]
```diff
@@ -3,6 +3,7 @@
 IMG ?= controller:latest
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 FLAGD_VERSION=v0.2.5
+CHART_VERSION=v0.2.6# x-release-please-version
```
Comment (beeme1mr) [inline Makefile:6]: Make sure you include this `Makefile` in the list of files Release Please needs to manage.

https://github.com/open-feature/open-feature-operator/blob/main/release-please-config.json#L9

File: chart/Chart.yaml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+type: application
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "v0.2.6" # x-release-please-version
```
Comment (beeme1mr) [inline chart/Chart.yaml:21]: Make sure you include this Makefile in the list of files Release Please needs to manage.

https://github.com/open-feature/open-feature-operator/blob/main/release-please-config.json#L9

File: chart/Chart.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (toddbaert) [reply chart/Chart.yaml:16]: We have numerous repos where release please is updating a semantic version number without a 'v'.

Check the pom file here: https://github.com/open-feature/java-sdk/blob/d67932052b012894e6432dabc613aa04b1ca102e/pom.xml#L7

I think what you have will work, this just seems simpler to me. Your call

**Comment by toddbaert** [comment]
I'll test this again in a couple hours when I'm back at my PC!

**Comment by toddbaert** [comment]
This all works for me provided I install cert-manager first, otherwise I'm missing objects.

```
Error: unable to build kubernetes objects from release manifest: [resource mapping not found for name: "ofo-serving-cert" namespace: "" from "": no matches for kind "Certificate" in version "cert-manager.io/v1"
```
I'm know this is a pre-requisite, and not a new one, but I'm still a bit confused if this hard dependency is a good idea. Could the cert manager requirement prevent adoption if an admin has some alternative policy or solution for generating self-signed certs? Is it more or less standard for operators that accept requests to require cert-manager? I'm not truly not sure, so I defer once more to your expertise, @AlexsJones !

File: Makefile (hunk 8 above)
Comment (toddbaert) [reply Makefile:6]: @AlexsJones I did this.

File: chart/Chart.yaml (hunk 9 above)
Comment (toddbaert) [reply chart/Chart.yaml:21]: @AlexsJones I did this.

**Comment by toddbaert** [comment]
@AlexsJones I pushed a commit to add a couple files to the release please `extra-files` member, which tells release please to look for the `x-release-please` annotations in those files. Hope you don't mind.

**Review by toddbaert — APPROVED** [review]

File: README.md [hunk 10]
```diff
@@ -19,15 +19,13 @@ _Requires [cert manager](https://cert-manager.io/docs/installation/kubernetes/)
 
 ```
 kubectl create namespace open-feature-operator-system
-kubectl apply -f https://github.com/open-feature/open-feature-operator/releases/download/v0.2.6/certificate.yaml
```
Comment (toddbaert) [inline README.md:22]: nice.

**Comment by AlexsJones** [comment]
> nfused if this hard dependency is a good idea. Could the cert manager requirement prevent adoption if an admin has some alternative policy or

For posterity, the reason it needs the certificate is for mTLS between the api-server and the webhook. 
It's something we do need, it doesn't have to be cert-manager, but given they just got brought into CNCF incubation I think we're backing the right horse

**Comment by AlexsJones** [comment]
Please review when you have a spare moment, I added to the readme @toddbaert 
Also I will squash all these commits, I was too lazy to rebase so did a skip

**Review by toddbaert — COMMENTED** [review]
Readme changes look good to me!

Sorry for that missing comma!

**Pull request merged**